### PR TITLE
Remove PopupMenu margin

### PR DESCRIPTION
**Pull request by HansMuller**
The margins make using showMenu's menuPosition argument difficult and they're not really needed.

I also made a few small gratuitous changes in navigator.dart.

### CODE REVIEW

**Comment by abarth** [comment]
Do we need to change showStockMenu so that the menu shows up in the same place on screen?

LGTM

**Comment by HansMuller** [comment]
Good point about showStockMenu. It's back to the way it was.

File: examples/stocks/lib/stock_menu.dart [hunk 1]
```diff
@@ -6,12 +6,14 @@ part of stocks;
...
+const double _kMenuMargin = 16.0; // 24.0 on tablet
+
 Future showStockMenu(NavigatorState navigator, { bool autorefresh, ValueChanged onAutorefreshChanged }) async {
   switch (await showMenu(
     navigator: navigator,
     position: new MenuPosition(
-      right: sky.view.paddingRight,
-      top: sky.view.paddingTop
+      right: sky.view.paddingRight + _kMenuMargin,
+      top: sky.view.paddingTop + _kMenuMargin
```
Comment (Hixie) [inline examples/stocks/lib/stock_menu.dart]: This shouldn't be pushed to the users. It's a material design spec margin, it should be implemented in the widgets. The author should just have to give the vague location that the menu should appear at ("top right", or "near this button", or some such) and we should apply the margins automatically.

Comment (abarth) [inline examples/stocks/lib/stock_menu.dart]: Maybe we should have a MenuPosition.topRight that has the right values...

Comment (HansMuller) [inline examples/stocks/lib/stock_menu.dart]: It seems like you want to say that the menu should appear the widget that triggered showing it and that it should additionally respect an additional margin around the application's body.

**Pull request merged**